[BUG] mypy tests fail in Windows

#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
`branch-3.0` on Windows 10

#### Issue:
`mypy tests` fails in `tests\test_examples.py` because `signal.alarm()` doesn't exist on Windows. This is preventing us from using Windows for the CI codebase checks, suggested in #11625 

#### Ideas...
- Remove `tests.test_examples` from `pyproject.toml` so it just gets skipped
- Refactor the `signal` stuff to separate file which isn't included in `pyproject.toml`
- Do away with `signal.alarm()`

Here's the console error output:
```
tests\test_examples.py: note: In function "_run_example":
tests\test_examples.py:287:23: error: Module has no attribute "SIGALRM"  [attr-defined]
            signal.signal(signal.SIGALRM, alarm_handler)
                          ^
tests\test_examples.py:288:9: error: Module has no attribute "alarm"  [attr-defined]
            signal.alarm(20 if not example.is_slow else 60)
            ^
tests\test_examples.py:301:13: error: Module has no attribute "alarm"  [attr-defined]
                signal.alarm(0)
                ^
Found 3 errors in 1 file (checked 295 source files)
```

--- a/.github/workflows/bokeh-ci.yml
+++ b/.github/workflows/bokeh-ci.yml
@@ -54,7 +54,12 @@ jobs:
           path: conda-bld-noarch.tgz
 
   codebase:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
 
     steps:
       - uses: actions/checkout@v2

--- a/bokeh/util/terminal.py
+++ b/bokeh/util/terminal.py
@@ -58,7 +58,8 @@ try:
     def white(text: str) -> str:  return "%s%s%s%s" % (Fore.WHITE, Style.BRIGHT, text, Style.RESET_ALL)
     def yellow(text: str) -> str: return "%s%s%s" % (Fore.YELLOW, text, Style.RESET_ALL)
 
-    sys.platform == "win32" and colorama.init()
+    if sys.platform == "win32":
+        colorama.init()
 except ImportError:
     def bright(text: str) -> str: return text
     def dim(text: str) -> str:    return text

--- a/release/ui.py
+++ b/release/ui.py
@@ -47,8 +47,8 @@ try:
     def yellow(text: str) -> str:
         return f"{colorama.Fore.YELLOW}{text}{colorama.Style.RESET_ALL}"
 
-    sys.platform == "win32" and colorama.init()
-
+    if sys.platform == "win32":
+        colorama.init()
 except ImportError:
 
     def bright(text: str) -> str:

--- a/tests/test_examples.py
+++ b/tests/test_examples.py
@@ -18,9 +18,9 @@ import pytest ; pytest
 
 # Standard library imports
 import os
-import platform
 import signal
 import subprocess
+import sys
 import time
 from os.path import (
     abspath,
@@ -62,8 +62,6 @@ from bokeh.util.terminal import (
 #-----------------------------------------------------------------------------
 # Setup
 #-----------------------------------------------------------------------------
-
-is_windows = platform.system() == "Windows"
 
 pytest_plugins = (
     "bokeh._testing.plugins.bokeh_server",
@@ -280,7 +278,7 @@ with open(filename, 'rb') as example:
     class Timeout(Exception):
         pass
 
-    if not is_windows:
+    if sys.platform != "win32":
         def alarm_handler(sig: int, frame: FrameType) -> NoReturn:
             raise Timeout
 
@@ -297,7 +295,7 @@ with open(filename, 'rb') as example:
         proc.kill()
         status = 'timeout'
     finally:
-        if not is_windows:
+        if sys.platform != "win32":
             signal.alarm(0)
 
     end = time.time()

